### PR TITLE
Python executable cleanups

### PR DESCRIPTION
**Pull request by Chroxvi**
This is a follow-up to #30 to fully implement [PEP394](https://peps.python.org/pep-0394/) in cotainr.

- Use `python3` executable instead of `python`.
- Use `sys.executable` when applicable
- Update docs to reflect that we implement PEP394
- Added an error early check to produce a meaningful error message when trying to run cotainr using a too old version of Python, i.e. instead of
  ```bash
  Traceback (most recent call last):
    File "./bin/cotainr", line 12, in <module>
      from cotainr.cli import main
    File "<fstring>", line 1
      (sub_cmd=)
              ^
  SyntaxError: invalid syntax
  ```
  you now get something like
  ```bash
  Cotainr requires Python>=3.8.0
  You are running Python==3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:53) 
  [GCC 9.4.0]
  from '/miniconda3/envs/py37_cotainr_test/bin/python3'
  ABORTING!
  ```

### CODE REVIEW

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

File: cotainr/__init__.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 """
 
+import sys
+
 __version__ = "2023.02.0"
```
Comment (eskech) [inline cotainr/__init__.py:12]: Are we waiting for the output change to create a new release?

**Review by eskech — APPROVED** [review]
Add documentation change also to state that we now only support python3

**Comment by Chroxvi** [comment]
> Add documentation change also to state that we now only support python3

We have always only supported python>=3.8 as stated on https://cotainr.readthedocs.io/en/latest/user_guide/index.html#cotainr-dependencies. The only change is that we now explicitly call `python3` instead of relying on `python` to be a symlink to `python3` as `python` may or may not exist and may or may not point to `python2` instead of `python3`.

File: cotainr/__init__.py (hunk 1 above)
Comment (Chroxvi) [reply cotainr/__init__.py:12]: Currently, we have #33, #34, and #36 staged for release when approved and merged. None of those seem critical to release ASAP, so I suggest we at least include any needed fix for #35 and possibly also a reworked version of #32 before making a new release.

**Pull request merged**